data/tokens/src/main/kotlin/com/tangem/data/tokens/utils/CryptoCurrencyConverter.kt

```diff
@@ -0,0 +1,52 @@
+package com.tangem.data.tokens.utils
+
+import com.tangem.blockchain.common.Blockchain
+import com.tangem.blockchain.common.Token as SdkToken
+import com.tangem.domain.common.DerivationStyleProvider
+import com.tangem.domain.tokens.models.CryptoCurrency
+import timber.log.Timber
+
+class CryptoCurrencyConverter {
+
+    fun createToken(
+        sdkToken: SdkToken,
+        blockchain: Blockchain,
+        derivationStyleProvider: DerivationStyleProvider,
+    ): CryptoCurrency.Token? {
+        if (blockchain == Blockchain.Unknown) {
+            Timber.e("Unable to map the SDK token to the domain token with Unknown blockchain")
+            return null
+        }
+
+        return CryptoCurrency.Token(
+            id = getTokenId(blockchain, sdkToken),
+            networkId = getNetworkId(blockchain),
+            name = sdkToken.name,
+            symbol = sdkToken.symbol,
+            iconUrl = getTokenIconUrl(blockchain, sdkToken),
+            decimals = sdkToken.decimals,
+            isCustom = false,
```

Comment:
Токен тут может быть кастомным, стоит это учитывать
```suggestion
            isCustom = isCustomToken(id),
```

Reply:
Исправил